src/components/SeatLayout/SeatLayout.tsx

```diff
@@ -0,0 +1,143 @@
+// />
+
+const SeatLayout = ({
+    layoutName,
+    seats,
+    selectedSeats,
+    updateSelectedSeats,
+}: {
+    layoutName: string;
+    seats: ISeat[];
+    selectedSeats: number[];
+    updateSelectedSeats: (seat: number) => void;
+}) => {
+    const [deck, setDeck] = useState<{
+        lowerDeck: number[][];
+    }>({ lowerDeck: [] });
+    const { t } = useTranslation('seatLayout');
+    let seatIndex = 0;
+
+    useEffect(() => {
+        setDeck((prev) => ({
+            ...prev,
+            lowerDeck: deckLayoutProducer(layoutConfig[layoutName].lowerDeck),
+        }));
+    }, [layoutName]);
+
+    return (
+        <SeatLayoutWrapper>
+            <div className="driver-cabin">
+                <img
+                    src={steeringWheel}
+                    alt="steering icon"
+                    className="steering"
+                />
+                <div className="line"></div>
+            </div>
+            <div className="seats-container">
+                {deck.lowerDeck.map((row: number[], rowIndex: number) => {
+                    return (
+                        <ul className="seat-row" key={`seatRow${rowIndex}`}>
+                            {row.map((seat: number, index: number) => {
+                                if (seat === 1) {
+                                    const currentSeat = seats[seatIndex];
+                                    seatIndex++;
+                                    return currentSeat.status.toString() ===
+                                        ISeatStatus.AVAILABLE.toString() ? (
+                                        selectedSeats.includes(
+                                            currentSeat.seatNumber
+                                        ) ? (
+                                            <Tooltip
+                                                title={`${
+                                                    currentSeat.seatNumber
+                                                }: ${t('selected')}`}
+                                                arrow
+                                                key={`seat${rowIndex}_${index}`}
+                                            >
+                                                <li
+                                                    className={`seat selected`}
+                                                    onClick={() => {
+                                                        updateSelectedSeats(
+                                                            currentSeat.seatNumber
+                                                        );
+                                                    }}
+                                                ></li>
+                                            </Tooltip>
+                                        ) : (
+                                            <Tooltip
+                                                title={`${
```

Comment:
can we optiimize this code, like single li/tooltip component and passing properties

Reply:
Changes done. Added a new component.